File: bigmler/static/out_model/items_analysis.js

    var escape = function(text) {
      return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
    };

    function itemMatches(text, fieldLabel, item) {
      /**
       * Computes item matches depending on the chosen item analysis options
       *
       * @param {string} text Input text
       * @param {string} fieldLabel Name of the field
       * @param {string} item Item to compare
       */

      var options = ITEM_ANALYSIS[fieldLabel];
      var separator = options.separator;
      var regexp = options.separator_regexp;
      if (typeof(regexp) === 'undefined') {
        if (typeof(separator) === 'undefined') {
          separator = " ";
        }
        regexp = escape(separator);
      }

      var pattern = new RegExp(regexp, 'g');
      var inputs = text.split(pattern);
      var counter = 0;
      for (var index = 0; index < inputs.length; index++) {
        if (inputs[index] == item) {
          counter++;
          break;
        }
      }
      return counter;
    };


    // predict body

File: bigmler/static/out_model/items_analysis.R

    escape <- function (text) {
      return(gsub("(\\W)", "\\\\\1", text))
    }

    itemMatches <- function (text, fieldLabel, item) {

      # Computes item matches depending on the chosen item analysis options
      #
      # @param {string} text Input text
      # @param {string} fieldLabel Name of the field
      # @param {string} item Item to compare

      if (is.na(text))
        return(0)
      options <- ITEM_ANALYSIS[[fieldLabel]]
      separator <- options[["separator"]]
      pattern <- options[["separator_regexp"]]
      if (is.null(pattern)) {
        if (is.null(separator)) {
          separator <- " "
        }
        pattern <- escape(separator)
      }
      inputs <- strsplit(text, pattern)
      if (item %in% inputs[[1]])
        return(1)
      return(0)
    }

    # predict function
